### ISSUE TITLE
Default audio stream preference is not obliged

### ISSUE THREAD

**Issue by mirh** [Sandmann79/xbmc#122]
Just installed this neat piece of software and all, and the first time I opened something I got czech dub (likely because it's the first alphabetically). 

Now, tbf I hadn't anything but the default Kodi settings, which means an english interface and "original stream's language" for player (so, maybe perhaps amazon isn't providing such datum).. But czech definitively shouldn't have been it.

**Comment by Varstahl** [comment]
The "original stream language" is not currently obliged for technical development reasons. It generally selects your preferred language, fallback to english, and if neither are available it just doesn't pre-select a specific audio track and Kodi does its thing, which probably yes, selects the first language available.

**Comment by mirh** [comment]
Well, both original stream, preferred language of default kody, and fallback.. are the same English language for Mr. Robot, and it seemingly didn't apply notwithstanding.

**Comment by Varstahl** [comment]
Ok, that is definitely weird and should not happen, I'm not aware of anyone else having this problem. If you still have troubles with it, could I take a look at your logfile? `%AppData%\Kodi\kodi.log` or the *nix equivalent `~/.kodi/temp/kodi.log`, you can post it somewhere like pastebin or the like.

**Comment by mirh** [comment]
https://pastebin.com/sWRSCg8a
This also applies to subtitles now btw (after having set them to All, from none)

**Comment by Varstahl** [comment]
È tutto molto strano. Non vedo problemi nei log, i sottotitoli vengono scaricati correttamente, la lingua viene rilevata correttamente dall'addon, ma non ti funziona. Mi domando se ci sia stato qualche regression in Kodi, appena posso provo ad aggiornare anch'io all'ultima nightly. Nel frattempo ho bisogno dei valori di 4 impostazioni, per capire:
* Amazon VOD Settings > General > Subtitles languages
* Kodi settings > Player > Language
    * Preferred audio language
    * Prefer default audio streams
    * Preferred subtitle language

**Comment by mirh** [comment]
All, original, on, original.

**Comment by Varstahl** [comment]
This explains a lot, given that the `original` settings are both not respected as of now, so the addon doesn't do anything except provide the streams, and Kodi rolls with it.

If you set things to `From settings, fallback to english`, Italian (or any other language), on, `Forced only` (or Italian/other language) everything should work as intended. You should also notice streams loading faster.

As for subtitles, Kodi picks the best (or what it thinks is best) out of the subtitles available to it. It is possible that switching subtitle stream in real time can break the timings for it, but that's a problem with Kodi itself and out of our control.

**Comment by mirh** [comment]
Makes sense, still I wonder if all those "subtitles languages" preferences are really needed?
Or are they just there to.. help testing/debugging or something?

**Comment by Varstahl** [comment]
We tried to please everyone. Since PrimeVideo has a plethora of language/subtitles combinations, and Kodi adds its bunch, we either decide what to do or cover most usage cases.

I wanted to provide more information on the settings, but Kodi doesn't allow contextual help, so… yeah.

**Comment by Varstahl** [comment]
Well, original is not really some option you'd set normally, at least in my case. Until it's English is all fine and dandy, but when you get some indian/arabic/dutch/whatever, I'm in deep trouble :D

I wanted to set up a multi-tier language "text box" with its own semantic to let the user control 100% what he wants and in what order, but it requires a lot of documentation and the user knowing exactly what it's doing. And, again, since Kodi provides absolutely nothing in the way of tooltips or context help, it's not a good idea.

**Comment by mirh** [comment]
Well, I also find original quite dumb in its way, but that's the default kodi uses after all. 
(and arguably the majority of people consumes english-native contents)

Anyway, if any (instead of banging your head over something that shouldn't be up to you) you could check out what the add-on I linked provides.

**Comment by Zuikkis** [comment]
@Varstahl could you explain in detail what settings I should use to get English audio? All shows are in German.

Only way I found, is select English as preferred language in Kodi settings. But this breaks so many other things I don't want to do that.

**Comment by Varstahl** [comment]
Pretty much selecting through Kodi audio language is the only way. Kodi has some limitations and don't really support multi language preferences. It's one of the things in the todo list, but I haven't been able to put time in the addon in the past year or so.
